### PR TITLE
Feature/list view feature list

### PR DESCRIPTION
**Pull request by xfeier53**
Implement the list view for the feature list. Now we can change the view by clicking the Tag button. Refactor on the structure a bit so that the filter button can be shared.

### CODE REVIEW

File: xbdd-web/src/modules/report/FeatureListContainer/TagViewFeatureList/styles/TagListStyles.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const tagListStyles = {
+  xbddTagList: {
+    borderRadius: "0 0 4px 4px",
+    height: "100%",
+    overflow: "scroll",
+    position: "relative",
+    width: "100%",
+  },
+  xbddFilterButton: {
+    width: 90,
```
Comment (rushton3179) [inline xbdd-web/src/modules/report/FeatureListContainer/TagViewFeatureList/styles/TagListStyles.js]: While this is valid can we change it to "90px" to be consistent

File: xbdd-web/src/modules/report/ReportContainer.jsx [hunk 2]
```diff
@@ -1,41 +1,33 @@
...
+    };
   }
 
   componentDidMount() {
-    getBuildReport(this.props.product, this.props.version, this.props.build).then(data => {
-      this.setState({
-        report: new Report(data),
-      });
-    });
+    // Fetch data for report or stat
```
Comment (rushton3179) [inline xbdd-web/src/modules/report/ReportContainer.jsx]: Is this going to be utilised or is it not getting used for anything? If you are going to put stuff in here later it's fine but let's delete it for now if you're not sure

File: xbdd-web/src/modules/report/FeatureListContainer/TagViewFeatureList/TagViewFeatureList.jsx [hunk 3]
```diff
@@ -15,20 +15,20 @@ const TagListFeaturesListView = props => {
 
   return (
     <List className={classes.xbddFeatureListContainer}>
-      {featureList.map(item => {
+      {featureList.map(feature => {
         return (
-          <ListItem button key={item.id} className={classesMap[item.calculatedStatus]}>
-            {item.name}
+          <ListItem button key={feature.id} className={classesMap[feature.calculatedStatus] + " " + classes.xbddTagViewFeatureList}>
```
Comment (rushton3179) [inline xbdd-web/src/modules/report/FeatureListContainer/TagViewFeatureList/TagViewFeatureList.jsx]: Can we use the backtick notation here for consistency? i.e. \`${classesMap[feature.calculatedStatus]} ${classes.xbddTagViewFeatureList}\`

File: xbdd-web/src/models/SimpleFeature.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class SimpleFeature {
+  constructor(data) {
+    this.id = data.id;
+    this.name = data.name;
+    this.calculatedStatus = data.calculatedStatus;
+    this.originalAutomatedStatus = data.originalAutomatedStatus;
+    this.tags = data.tags;
+    this.restId = data["_id"];
```
[the remark this answers is not in the document]
Comment (xfeier53) [reply xbdd-web/src/models/SimpleFeature.js]: I did that previously but got an error: Unexpected dangling '_' in '_id'.eslint(no-underscore-dangle)

File: xbdd-web/src/modules/report/FeatureListContainer/TagViewFeatureList/styles/TagListStyles.js (hunk 1 above)
Comment (xfeier53) [reply xbdd-web/src/modules/report/FeatureListContainer/TagViewFeatureList/styles/TagListStyles.js]: Just found that it is not used at all, so I deleted it.

File: xbdd-web/src/modules/report/ReportContainer.jsx (hunk 2 above)
Comment (xfeier53) [reply xbdd-web/src/modules/report/ReportContainer.jsx]: Yes, the next thing is to handle when a feature is clicked here and render the report next to the feature list.

**Pull request merged**